master/buildbot/changes/gitpoller.py

```diff
@@ -236,6 +236,8 @@ def poll(self):
                 log.err(_why="trying to poll branch {} of {}".format(
                         branch, self.repourl))
 
+        # replace self.lastRev contents with revs
+        self.lastRev.clear()
         self.lastRev.update(revs)
```

Comment:
Is there a reason why we can't just set lastRev to revs?

Reply:
Probably not. I've changed it in 01bb6f5269e503bb46ea4288dceadc42e04eff68.